### PR TITLE
env_process: Introduce a new param to configure the kernel command line

### PR DESCRIPTION
**Pull request by PaulYuuu**
Introduce a new param to configure the kernel command line

Introduce kernel_extra_params_serial_login to configure the kernel command line in env_process, sometimes the guest does not have a serial session available. The issue is from #2564

ID: 1899377
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
Hello @chloerh @kylazhang, this problem will cause the case "virsh.console" report "No console available" when "update_console = yes".

Hello @vivianQizhu @quanwenli, some guests got "[   40.664424] TCP: lp registered" output in the serial log during the final stage of boot, so serial session can not handle it.
```
08:13:05 ERROR|     raise ExpectTimeoutError(patterns, output)
08:13:05 ERROR| aexpect.exceptions.ExpectTimeoutError: Timeout expired while looking for pattern '^\\[.*\\][\\#\\$]\\s*$'    (output: 'BOOT_IMAGE=/vmlinuz-3.10.0-1160.el7.ppc64le root=/dev/mapper/rhel_dhcp16--215--202-root ro console=ttyS0,115200 crashkernel=auto rd.lvm.lv=rhel_dhcp16-215-202/root rd.lvm.lv=rhel_dhcp16-215-202/swap biosdevname=0 net.ifnames=0 console=tty0 LANG=en_US.UTF-8 movable_node\n[root@dhcp16-215-226 ~]# [   27.438810] fuse init (API version 7.23)\n[   40.664424] TCP: lp registered\n')
08:13:05 ERROR| 
```

**Comment by vivianQizhu** [comment]
@PaulYuuu I don't fully understand the issue, would you share the complete test log somewhere? Thanks.

**Comment by PaulYuuu** [comment]
> @PaulYuuu I don't fully understand the issue, would you share the complete test log somewhere? Thanks.

@vivianQizhu I filed a bug to track it. Please see the bug for details on the qemu part.

File: virttest/env_process.py [hunk 1]
```diff
@@ -303,6 +303,7 @@ def preprocess_vm(test, params, env, name):
                           migration_exec_cmd=params.get("migration_exec_cmd_dst"))
 
         # Update kernel param
+        serial_update = params.get_boolean("kernel_extra_params_serial_update")
```
Comment (vivianQizhu) [inline virttest/env_process.py:306]: Recommend to name it `update_boot_option_serial_login`.

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -166,7 +166,7 @@ def __run_cmd_and_handle_error(msg, cmd, session, test_fail_msg):
 
 
 def update_boot_option(vm, args_removed="", args_added="",
-                       need_reboot=True, guest_arch_name='x86_64'):
+                       need_reboot=True, guest_arch_name='x86_64', serial=False):
```
Comment (vivianQizhu) [inline virttest/utils_test/__init__.py:169]: Add here `serial_login=False`

**Review by vivianQizhu — COMMENTED** [review]
LGTM otherwise. @quanwenli Request for your review, it might require you update your test case with certain params specification if we merge this.

File: virttest/utils_test/__init__.py (hunk 2 above)
Comment (quanwenli) [reply virttest/utils_test/__init__.py:169]: @vivianQizhu Do you mean I need to adding "kernel_extra_params_serial_update = yes" in virtio_net_dpdk.cfg ?

Comment (vivianQizhu) [reply virttest/utils_test/__init__.py:169]: @quanwenli I guess so, if we have this in.

File: virttest/env_process.py (hunk 1 above)
Comment (PaulYuuu) [reply virttest/env_process.py:306]: The `kernel_extra_params` prefix is ​​used to maintain alignment with `kernel_extra_params_add` and `kernel_extra_params_remove`. `update_boot_option` is a method we invoked but not very suitable as a param name in `env_process`. How about keeping this one or using `kernel_extra_params_serial_login`?

File: virttest/utils_test/__init__.py (hunk 2 above)
Comment (sathnaga) [reply virttest/utils_test/__init__.py:169]: @vivianQizhu @PaulYuuu current method defaults serial=True and why change it to False bydefault?
we can leave the default setting to true, as we anyway control it via param in other places needed?

File: virttest/env_process.py (hunk 1 above)
Comment (vivianQizhu) [reply virttest/env_process.py:306]: > The `kernel_extra_params` prefix is ​​used to maintain alignment with `kernel_extra_params_add` and `kernel_extra_params_remove`. `update_boot_option` is a method we invoked but not very suitable as a param name in `env_process`. How about keeping this one or using `kernel_extra_params_serial_login`?

Okay, make sense.

File: virttest/utils_test/__init__.py (hunk 2 above)
Comment (vivianQizhu) [reply virttest/utils_test/__init__.py:169]: > @vivianQizhu @PaulYuuu current method defaults serial=True and why change it to False bydefault?
> we can leave the default setting to true, as we anyway control it via param in other places needed?

@sathnaga Thanks for your review. Actually, previously it's `False` by default, we just introduced serial login earlier this year here: https://github.com/avocado-framework/avocado-vt/pull/2564 for a specific network test case, for now only this case need `serial=True` and other test cases is okay with `False` by default.

Comment (PaulYuuu) [reply virttest/utils_test/__init__.py:169]: @sathnaga Changing the default value to False is due to some issues with serial login, please see https://github.com/avocado-framework/avocado-vt/pull/2828#issuecomment-729715574

**Review by vivianQizhu — APPROVED** [review]

**Comment by vivianQizhu** [comment]
@quanwenli Request for your approval, thanks.

**Comment by quanwenli** [comment]
LGTM

File: virttest/utils_test/__init__.py (hunk 2 above)
Comment (vivianQizhu) [reply virttest/utils_test/__init__.py]: @sathnaga Any more comments? Thanks.

Comment (sathnaga) [reply virttest/utils_test/__init__.py]: @vivianQizhu LGTM,

**Review by sathnaga — APPROVED** [review]

**Comment by vivianQizhu** [comment]
Thanks all.

**Pull request merged**